duckduckgo/Android
# Add new pixels for android instrumentation

Task/Issue URL: https://app.asana.com/0/361428290920652/1119886581463189
Tech Design URL: 

**Description**:
TIdy up pixels to for better widgets and default browser counts. Also add pixel to detect widget engagement.

**Steps to test this PR**:

**TEST 1: Add / Remove widget Test**
1. Run the app add a bunch of widgets
1. Note the add widget pixel "m_w_a" is fired with lang and country only once
1. Launch the app from the widget and check that the "m_w_l" pixel launches with lang and country
1. Delete all the widgets
1. Note the delete widget pixel "m_w_d" is fired once when the last widget is removed
1. Check that other pixels do not contain lang and country

**TEST 2: Set / Unset default browser** 
1. From settings, set the app as the default browser
1. Launch the app
1. Note the default browser pixel "m_db_s" is fired with lang and country
1. From settings, choose a different app to be the default browser
1. Note the unset pixel "m_db_u" is fired with lang and country
1. Check that other pixels do not contain lang and country

**TEST 3: Device Restart**
1. Restart device
1. Ensure pixels are not resent unless newly triggered 

**TEST 4: Onboarding**
1. Freshly install the app
1. When onboarding select option to set as default browser
1. Ensure pixel sent

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
